src/compiler/checker.ts

```diff
@@ -20255,7 +20308,78 @@ namespace ts {
             return isCallOrNewExpression(node.parent) && node.parent.expression === node;
         }
 
-        function checkPropertyAccessExpressionOrQualifiedName(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, right: Identifier) {
+        function getPropertyForPrivateIdentifier(leftType: Type, right: PrivateIdentifier, errorNode: Node | undefined): Symbol | undefined {

```

Comment:
Would you be able to re-review `getPropertyForPrivateIdentifier`?

I rewrote the getPropertyForPrivateIdentifier function to:
- add related spans per @DanielRosenwasser's feedback
- fix a bug with shadowing detection (see privateNamesInNestedClasses-2.ts)
- improve readability by using `findAncestor`